subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/DefaultProjectConnection.java

```diff
@@ -63,16 +70,47 @@ public TestLauncher newTestLauncher() {
         if (!modelType.isInterface()) {
             throw new IllegalArgumentException(String.format("Cannot fetch a model of type '%s' as this type is not an interface.", modelType.getName()));
         }
-        return new DefaultModelBuilder<T>(modelType, connection, parameters);
+        return new DefaultModelBuilder<>(modelType, connection, parameters);
     }
 
     @Override
     public <T> BuildActionExecuter<T> action(final BuildAction<T> buildAction) {
-        return new DefaultBuildActionExecuter<T>(buildAction, connection, parameters);
+        return new DefaultBuildActionExecuter<>(buildAction, connection, parameters);
     }
 
     @Override
     public BuildActionExecuter.Builder action() {
         return new DefaultBuildActionExecuter.Builder(connection, parameters);
     }
+
+    @Override
+    public void notifyDaemonsAboutChangedPaths(List<Path> changedPaths) {
+        List<String> absolutePaths = new ArrayList<>();
```

Comment:
Let's allocate this to the size of `changedPaths`.

Reply:
Done.